services/QuillLMS/client/app/bundles/Teacher/components/unit_templates/unit_template_activity_selector.tsx

```diff
@@ -0,0 +1,288 @@
+import * as React from 'react'
+import request from 'request';
+
+import UnitTemplateActivityDataRow from './unit_template_activity_data_row'
+
+import { FlagDropdown, SortableList } from '../../../Shared/index'
+import Pagination from '../../../Teacher/components/assignment_flow/create_unit/custom_activity_pack/pagination'
+import { lowerBound, upperBound, } from '../../../Teacher/components/assignment_flow/create_unit/custom_activity_pack/shared'
+
+const ACTIVITIES_URL = `${process.env.DEFAULT_URL}/activities/index_with_unit_templates`
+const DEFAULT_FLAG = 'All Flags'
+const DEFAULT_TOOL = 'All Tools'
+const DEFAULT_READABILITY = 'All readability levels'
+const TOOL_OPTIONS = [
+  'All Tools',
+  'Quill Connect',
+  'Quill Grammar',
+  'Quill Proofreader',
+  'Quill Evidence',
+  'Quill Lessons',
+  'Quill Diagnostic'
+]
+const UNSELECTED_TYPE = 'unselected'
+const SELECTED_TYPE = 'selected'
+
+const UnitTemplateActivitySelector = ({ parentActivities, setParentActivities, toggleParentActivity }) => {
+
+  const [activities, setActivities] = React.useState([])
+  const [selectedActivities, setSelectedActivities] = React.useState(parentActivities)
+  const [loading, setLoading] = React.useState(true);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [nameSearch, setNameSearch] = React.useState('')
+  const [descriptionSearch, setDescriptionSearch] = React.useState('')
+  const [ccssSearch, setCCSSSearch] = React.useState('')
+  const [conceptSearch, setConceptSearch] = React.useState('')
+  const [activityPacksSearch, setActivityPacksSearch] = React.useState('')
+  const [flagSearch, setFlagSearch] = React.useState(DEFAULT_FLAG)
+  const [toolSearch, setToolSearch] = React.useState(DEFAULT_TOOL)
+  const [readabilitySearch, setReadabilitySearch] = React.useState(DEFAULT_READABILITY)
+  const [readabilityOptions, setReadabilityOptions] = React.useState([])
+  const [showNoActivityPacks, setShowNoActivityPacks] = React.useState(false)
```

Comment:
just for the sake of pattern, it would be nice to have some type definitions for these